tests/e2e/vmc_delete_tkg.go

```diff
@@ -0,0 +1,272 @@
+	f := framework.NewDefaultFramework("vmc-delete-tkg")
+	var (
+		client            clientset.Interface
+		svNamespace       string
+		scParameters      map[string]string
+		storagePolicyName string
+		deleteGC          bool
+
+		vmcUser string
+	)
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		svcClient, svNamespace = getSvcClientAndNamespace()
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		bootstrap()
+		setResourceQuota(svcClient, svNamespace, rqLimit)
+		scParameters = make(map[string]string)
+	})
+	ginkgo.AfterEach(func() {
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+	})
+
+	/*
+		Test to Delete TKG on VMC
+		Steps
+			1. Create SC
+			2. Create PVC with the above created SC and validate PVC is in bound phase
+			3. Create POD with the above-created PVC and validate pod is in running state
+			4. Delete the New Guest Cluster and validate the deleted Guest Cluster is not accessible
+			5. Verify Volume Handle is available in SVC for the PVC created in the deleted GC
+			6. Create PV in existing GC using the volume handle and validate it is in the available status
+			7. Create PVC using the PV created in above step and validate it is in bound status
+			8. Create POD validate it is in running status
+			9. Verify CRD in SVC
+			10. Delete POD, PVC, and PV in GC
+	*/
+
+	ginkgo.It("[delete-tkg-vmc] Delete TKG", func() {
+		var err error
+		var wcpToken string
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		deleteGC = false
+
+		newGcKubconfigPath := os.Getenv("DELETE_GUEST_CLUSTER_KUBE_CONFIG")
+		if newGcKubconfigPath == "" {
+			ginkgo.Skip("Env DELETE_GUEST_CLUSTER_KUBE_CONFIG is missing")
+		}
+
+		vmcUser = os.Getenv("VMC_USER")
+		if vmcUser == "" {
+			ginkgo.Skip("Env VMC_USER is missing")
+		}
+
+		tkg_cluster := os.Getenv("TKG_CLUSTER_TO_DELETE")
+		if tkg_cluster == "" {
+			ginkgo.Skip("Env TKG_CLUSTER_TO_DELETE is missing")
+		}
+
+		clientNewGc, err := k8s.CreateKubernetesClientFromConfig(newGcKubconfigPath)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+			fmt.Sprintf("Error creating k8s client with %v: %v", newGcKubconfigPath, err))
+
+		ginkgo.By("Creating namespace on GC2")
+		ns, err := framework.CreateTestingNS(f.BaseName, clientNewGc, map[string]string{
+			"e2e-framework": f.BaseName,
+		})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Error creating namespace on GC2")
+
+		namespaceNewGC := ns.Name
+		framework.Logf("Created namespace on GC2 %v", namespaceNewGC)
+		defer func() {
+			if !deleteGC {
+				err := clientNewGc.CoreV1().Namespaces().Delete(ctx, namespaceNewGC, *metav1.NewDeleteOptions(0))
```

Comment:
This operation used to through some exception if the namespace takes longer time to delete. 
You can use this poll wait method till this operation completes 
 waitForNamespaceToGetDeleted() , you can use 

ex: framework.ExpectNoError(waitForNamespaceToGetDeleted(ctx,
			client, namespaceToDelete, poll, supervisorClusterOperationsTimeout))